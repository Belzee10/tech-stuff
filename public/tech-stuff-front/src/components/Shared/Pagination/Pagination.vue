<template>
  <v-pagination
    :value="value"
    :length="length"
    :total-visible="totalVisible"
    @input="v => $emit('input', v)"
    @next="$emit('next')"
    @previous="$emit('previous')"
  ></v-pagination>
</template>

<script>
export default {
  name: 'Pagination',
  props: {
    length: {
      type: Number,
      default: 0
    },
    totalVisible: {
      type: Number,
      default: 0
    },
    value: {
      type: Number,
      default: 0
    }
  }
};
</script>

<style lang="scss" scoped></style>
