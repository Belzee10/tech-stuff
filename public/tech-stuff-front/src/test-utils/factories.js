import { mount, createLocalVue } from '@vue/test-utils';
import mergeWith from 'lodash.mergewith';
import Vuetify from 'vuetify';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);

let vuetify;

beforeEach(() => {
  vuetify = new Vuetify();
});

const customizer = (_, srcValue) => {
  if (Array.isArray(srcValue)) {
    return srcValue;
  }
  if (srcValue instanceof Object && Object.keys(srcValue).length === 0) {
    return srcValue;
  }
};

/**
 * create component wrapper
 * @param {*} Component
 * @param {Object} overrides
 */
const createWrapper = (Component, overrides) => {
  const defaultMountingOptions = {
    localVue,
    vuetify
  };
  return mount(
    Component,
    mergeWith(defaultMountingOptions, overrides, customizer)
  );
};

/**
 * create Vuex store
 * @param {Object} overrides
 */
const createStore = overrides => {
  const defaultStoreConfig = {};
  return new Vuex.Store(mergeWith(defaultStoreConfig, overrides, customizer));
};

export { createWrapper, createStore };
