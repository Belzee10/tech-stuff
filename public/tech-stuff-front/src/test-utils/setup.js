import Vue from 'vue';
import Vuetify from 'vuetify';

Vue.config.productionTip = false;

Vue.use(Vuetify);

/**
 * silent "console.error"
 */
global.console = {
  error: jest.fn(),
  log: console.log,
  warn: console.warn,
  info: console.info,
  debug: console.debug
};
