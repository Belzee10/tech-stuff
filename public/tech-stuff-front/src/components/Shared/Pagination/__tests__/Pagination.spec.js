import { createWrapper } from '@/test-utils/factories';
import Pagination from '../Pagination.vue';

describe('Pagination.vue', () => {
  test('should ', () => {
    const wrapper = createWrapper(Pagination);
    expect(wrapper.element).toMatchSnapshot();
  });
});
