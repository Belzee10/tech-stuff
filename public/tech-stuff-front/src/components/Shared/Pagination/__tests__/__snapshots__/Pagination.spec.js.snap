// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination.vue should  1`] = `
<ul
  class="v-pagination theme--light"
>
  <li>
    <button
      class="v-pagination__navigation v-pagination__navigation--disabled"
      type="button"
    >
      <i
        aria-hidden="true"
        class="v-icon notranslate mdi mdi-chevron-left theme--light"
      />
    </button>
  </li>
  <li>
    <button
      class="v-pagination__navigation v-pagination__navigation--disabled"
      type="button"
    >
      <i
        aria-hidden="true"
        class="v-icon notranslate mdi mdi-chevron-right theme--light"
      />
    </button>
  </li>
</ul>
`;
